tests/mapped_struct.py

```diff
@@ -1500,6 +1500,27 @@ def testProxiedListRepr(self):
     def testProxiedListSpecificEqual(self):
         self.assertEquals(self.pack([1, 2.0]), (1, 2.0))
 
+    def testProxiedListSlice(self):
+        orig = [(str(i) if (i % 2) == 0 else i) for i in range(20)]
+        obj = self.pack(orig)
+        huge = 1 << 40
+
+        self.assertEquals(obj[1:], orig[1:])
+        self.assertEquals(obj[:-1], orig[:-1])
+        self.assertEquals(obj[2:4], orig[2:4])
+        self.assertEquals(obj[1:-1:2], orig[1:-1:2])
+        self.assertEquals(obj[::1], orig[::1])
+        self.assertEquals(obj[::2], orig[::2])
+        self.assertEquals(obj[::-1], orig[::-1])
+        self.assertEquals(obj[1::-1], orig[1::-1])
+        self.assertEquals(obj[2:][:5], orig[2:][:5])
+        self.assertEquals(obj[::3][::2], orig[::3][::2])
+        self.assertEquals(obj[1:10:2][2::5], orig[1:10:2][2::5])
+        self.assertEquals(obj[10:1:-3][2::2], orig[10:1:-3][2::2])
+        self.assertEquals(obj[huge:huge], orig[huge:huge])
+        self.assertEquals(obj[huge:huge:huge], orig[huge:huge:huge])
```

Comment:
Add tests that test out of bounds access on slices (ie, the IndexError s). This equals thing will only test within-bounds tests.

Also add some "NotEquals" assertions to make sure the test fails if somehow the comparator is bogus.

Reply:
You're missing the NotEqual